docs/using.md

```diff
@@ -0,0 +1,282 @@
+# Using Quix Streams
+
+In this topic you will learn how to use Quix Streams to perform two types of data processing:
+
+1. **One message at a time processing** - Here the message received contains all required data for processing. No state needs to be preserved between messages, or between replicas. The data from the message is used to calculate a new value, which is then typically published to the output stream.
+2. **Stateful processing** - This is where you need to keep track of data between messages, such as keeping a running total of a variable. This is more complicated as state needs to be preserved between messages, and potentially between replicas, where multiple replicas are deployed. In addition, state may need to be preserved in the event of the failure of a deployment - Quix Streams supports checkpointing as a way to enable this.
+
+The following sections will explore these methods of data processing in more detail.
+
+## Topics, streams, partitions, replicas, and consumer groups
+
+The main structure used for data organization in Quix is the topic. For example, the topic might be `iot-telemetry`. To allow for horizontal scaling, a topic is typically divided into multiple streams. You may have multiple devices, or sources, writing data into a topic, so to ensure scaling and message ordering, each source writes into its own stream. Device 1 would write to stream 1, and device 2 to stream 2 and so on. This is the idea of [stream context](./features/streaming-context.md). 
```

Comment:
I would add the following paragraph: If you wish to aggregate data beyond the stream context of a source, slice streams in the pipeline based on a different key. For example, invoices are being streamed from stores where each store has one stream. We want to aggregate items per StockCode to see how much of each article was sold. We need to first split invoices into streams per StockCode and then perform aggregation.